apple/swift-docc
# Support resolving documentation links to content in other DocC archives

Bug/issue #, if applicable: rdar://114731067

## Summary

This adds experimental support for resolving documentation links and symbol links to content in other DocC archives.

At a high level this external link support works in two steps:
- While building the documentation for the first module, DocC emits two files, one that describe the link hierarchy and its possible disambiguations and one that contains minimal amount of content for every linkable page.
- Later, while building the documentation for the second module, DocC is passed the documentation archive for the first module and reads these file and uses them to resolve links to content in the first archive.



> [!IMPORTANT]
> There is more work needed to support this end-to-end. If you run this now you'll notice that the external links are 404s in the browser unless you host both archives on the same server.

## Dependencies

None

## Testing

To test this you need a project with two documentable targets.

- Build documentation for one of the targets and pass `--enable-experimental-external-link-support` to enable linking to this documentation.

- Verify that the output directory contains a "linkable-entities.json" file and a "link-hierarchy.json" file.

- In the other target, add links to symbols in the first target starting with a leading slash and the name of the first module, for example `/FirstModule/MyClass/myFunction()`

- Build documentation for the other target and pass `--dependency /path/to/first-target.doccarchive`.

- The link should resolve to display the documentation comment for `myFunction()` from the first target.

- Change the external documentation link to introduce a typo. The external link should raise an error with a fix-it, same as it would for a local link.

- Remove the "linkable-entities.json" a "link-hierarchy.json" files from the first archive and build documentation for the second target again. 

- There should be an error about missing files in the dependency.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
